spylon/spark/yarn_launcher.py

```diff
@@ -262,6 +265,24 @@ def launcher(deploy_mode, args, working_dir="."):
     i = spark_args.index("--conda-env")
     conda_env = spark_args[i+1]
     spark_args = spark_args[:i] + spark_args[i+2:]
+    cleanup_functions = []
+
+    def extract_local_archive(_env_name, _local_archive):
+        """Helper internal function for extracting a zipfile and ensure that a cleanup is queued."""
+        with zipfile.ZipFile(_local_archive) as z:
+            z.extractall(working_dir)
+            archive_filenames = z.namelist()
+
+        def cleanup():
+            for fn in [os.path.abspath(os.path.join(working_dir, f)) for f in archive_filenames]:
```

Comment:
Move the abspath out of cleanup so that it captures the meaning of "." when the function runs, not when cleanup runs.